Sources/SwiftStylable/Classes/Style/Colors/ColorCollection.swift

```diff
@@ -33,14 +33,15 @@ public class ColorCollection {
             for (name, colorString) in colorEntries {
                 if let color = UIColor(hexString: colorString) {
                     if let colorHolder = self._colorHolders[name] {
+                        colorHolder.reference = nil
                         colorHolder.color = color
                     } else {
                         self._colorHolders[name] = ColorHolder(color: color)
                     }
                     colorEntries.removeValue(forKey: name)
                     numParsedColors += 1
                 } else if let colorHolder = self._colorHolders[colorString] {
-                    self._colorHolders[name] = colorHolder
+                    self._colorHolders[name] = ColorHolder(reference: colorHolder)
```

Comment:
Hier twijfel ik over: zou je niet hier ook moeten checken of er al een holder is voor deze name en als dat zo is alleen de reference moeten setten?

Reply:
Nee volgens mij juist niet. Als de key er is wil je altijd dat hij een nieuwe Holder aanmaakt, anders zou hij er niet staan. Als je een key gebruikt moet hij altijd eerdere keys overschrijven. name = key in plist.

Reply:
Ja, maar je hebt nou juist die reference aangemaakt zodat je de holder NIET hoeft te overschrijven. Je wilt de waarde IN de holder overschrijven met een nieuwe referentie. als je een nieuwe holder aanmaakt, zullen bestaande components met een verwijzing naar die kleur hun verwijzing kwijt raken en dus (misschien) niet updaten.

Reply:
Pff doet een beetje pijn in mijn hoofd maar ik denk dat je gelijk hebt. Vind jet het dan logisch dat ik de size op nil zet?

Reply:
Ja de size / color kan dan op nil, dat is oude informatie. Als je het echt mooi wilt doen zou je bijv ook de color, size en reference properties read only kunnen maken en functies maken set(with color: UIColor) en set(with reference: ColorHolder). Die functies zetten dan elk de andere optie op nil. Dit kunnen de setters ook gewoon doen, maar op deze manier is het iets explicieter dat je niet bijv. enkel de color set.

Reply:
Ik zal eens kijken